### PR TITLE
feat(arduino-cli): add completion spec

### PR DESCRIPTION
**Pull request by lukejans**
## arduino-cli completion spec 
- arduino-cli documentation used for spec: [link](https://arduino.github.io/arduino-cli/0.35/commands/arduino-cli/)

Screenshot 2024-01-12 at 6 20 18 PM

### CODE REVIEW

**Comment by lukejans** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mschrage** [comment]
Hi @lukejans! Love this contribution, thanks for submitting - can you remove the yarn.lock file from this PR, so we can get this merged in?

**Comment by grant0417** [comment]
Sorry, to clarify don't delete the `yarn.lock` just revert the changes to it with `git fetch && git checkout master -- yarn.lock`.

File: src/arduino-cli.ts [hunk 1]
```diff
@@ -0,0 +1,1661 @@
...
+    } catch (error) {
+      return [];
+    }
+  },
+};
+
+const completionSpec: Fig.Spec = {
+  name: "arduino-cli",
+  description:
+    "Arduino Command Line Interface (arduino-cli), example: 'arduino-cli <command> [flags...]'",
```
Comment (grant0417) [inline src/arduino-cli.ts:43]:
```suggestion
    "Arduino Command Line Interface",
```

File: src/arduino-cli.ts [hunk 2]
```diff
@@ -0,0 +1,1661 @@
...
+  subcommands: [
+    {
+      name: "board",
+      description:
+        "Arduino board commands, example: 'arduino-cli board <command> [flags]'",
```
Comment (grant0417) [inline src/arduino-cli.ts:48]:
```suggestion
        "Arduino board commands",
```

File: src/arduino-cli.ts [hunk 3]
```diff
@@ -0,0 +1,1661 @@
...
+      args: {
+        name: "attach, list...",
+        isOptional: true,
+      },
+      subcommands: [
+        {
+          name: "attach",
+          description:
+            "Sets the default values for port and FQBN. If no port or FQBN are specified, the current default port and FQBN are displayed, example: 'arduino-cli board attach [-p <port>] [-b <FQBN>] [sketchPath] [flags]'",
```
Comment (grant0417) [inline src/arduino-cli.ts:57]:
```suggestion
            "Sets the default values for port and FQBN. If no port or FQBN are specified, the current default port and FQBN are displayed",
```

File: src/arduino-cli.ts [hunk 4]
```diff
@@ -0,0 +1,1661 @@
...
+              args: {
+                name: "protocol",
+              },
+            },
+          ],
+        },
+        {
+          name: "details",
+          description:
+            "Show information about a board, in particular if the board has options to be specified in the FQBN, example: 'arduino-cli board details -b <FQBN> [flags]'",
```
Comment (grant0417) [inline src/arduino-cli.ts:109]:
```suggestion
            "Show information about a board, in particular if the board has options to be specified in the FQBN'",
```

File: src/arduino-cli.ts [hunk 5]
```diff
@@ -0,0 +1,1661 @@
...
+            {
+              name: "--board-options",
+              description:
+                "List of board options separated by commas. Or can be used multiple times for multiple options",
+              args: {},
+            },
+            {
+              name: "--discovery-timeout",
+              description:
+                "Max time to wait for port discovery, example: '30s, 1m (default 1s)'",
```
Comment (grant0417) [inline src/arduino-cli.ts:68]: This is fine

File: src/arduino-cli.ts [hunk 6]
```diff
@@ -0,0 +1,1661 @@
...
+              args: {
+                name: "duration",
+              },
+            },
+            {
+              name: ["-b", "--fqbn"],
+              description:
+                "Fully Qualified Board Name, example: 'arduino:avr:uno'",
```
Comment (grant0417) [inline src/arduino-cli.ts:76]: This is fine

File: src/arduino-cli.ts (hunk 3 above)
Comment (grant0417) [reply src/arduino-cli.ts:57]: This has the full cli command

File: src/arduino-cli.ts (hunk 2 above)
Comment (grant0417) [reply src/arduino-cli.ts:48]: This has the full cli command

File: src/arduino-cli.ts (hunk 1 above)
Comment (grant0417) [reply src/arduino-cli.ts:43]: This has the full cli command

**Review by grant0417 — CHANGES_REQUESTED** [review]
Spec looks good, one change I would suggest before merge is to remove all the `example` in description where it is the full CLI command. Left some comments as examples.

**Review by grant0417 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**